std/algebra/emulated/sw_bw6761/pairing_test.go

```diff
@@ -0,0 +1,253 @@
+package sw_bw6761
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"testing"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	bw6761 "github.com/consensys/gnark-crypto/ecc/bw6-761"
+	"github.com/consensys/gnark/constraint"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/frontend/cs/r1cs"
+	"github.com/consensys/gnark/frontend/cs/scs"
+	"github.com/consensys/gnark/test"
+)
+
+func randomG1G2Affines() (bw6761.G1Affine, bw6761.G2Affine) {
+	_, _, G1AffGen, G2AffGen := bw6761.Generators()
+	mod := bw6761.ID.ScalarField()
+	s1, err := rand.Int(rand.Reader, mod)
+	if err != nil {
+		panic(err)
+	}
+	s2, err := rand.Int(rand.Reader, mod)
+	if err != nil {
+		panic(err)
+	}
+	var p bw6761.G1Affine
+	p.ScalarMultiplication(&G1AffGen, s1)
+	var q bw6761.G2Affine
+	q.ScalarMultiplication(&G2AffGen, s2)
+	return p, q
+}
+
+type FinalExponentiationCircuit struct {
+	InGt GTEl
+	Res  GTEl
+}
+
+func (c *FinalExponentiationCircuit) Define(api frontend.API) error {
+	pairing, err := NewPairing(api)
+	if err != nil {
+		return fmt.Errorf("new pairing: %w", err)
+	}
+	res := pairing.FinalExponentiation(&c.InGt)
+	// gnark-crypto exponent is 12(x₀+1) and gnark is (x₀-1)
+	tmp := pairing.Expc1(res)
```

Comment:
Don't understand the need for computing this. Can we inline to FinalExp or AssertIsEqual?

Reply:
This is a temporary hack to test the new final exp. There is currently a discrepancy between gnark and gnark-crypto results (a factor 12 in the exponent), but both results are correct as they define a bilinear map (similarly to [this](https://hackmd.io/@yelhousni/bn254-pairing-fuzzer)). It should be soon resolved after implementing this issue https://github.com/Consensys/gnark-crypto/issues/457.